### ISSUE TITLE
pythoncapi_compat tool, C API updates

### ISSUE THREAD

**Issue by illume** [pygame/pygame#3018]
https://github.com/pythoncapi/pythoncapi_compat

Started in https://github.com/pygame/pygame/pull/3014 there are some remaining items left to do with Py_Is.

See https://github.com/pygame/pygame/pull/3014 where all the other parts were fixed.

**Comment by Starbuck5** [comment]
Yeah, the only remaining thing to do are the `Py_Is` function, which I don't really see a need to do. They're new in 3.10, so we can give it a while.
